src/lib/libwasmfs.js

```diff
@@ -195,13 +195,14 @@ addToLibrary({
       } else {
         bytesRead = __wasmfs_read(stream.fd, dataBuffer, length);
       }
-      bytesRead = FS.handleError(bytesRead);
-
-      for (var i = 0; i < length; i++) {
-        buffer[offset + i] = {{{ makeGetValue('dataBuffer', 'i', 'i8')}}}
+      if (bytesRead > 0) {
+        for (var i = 0; i < bytesRead; i++) {
+          buffer[offset + i] = {{{ makeGetValue('dataBuffer', 'i', 'i8')}}}
+        }
```

Comment:
```suggestion
      for (var i = 0; i < bytesRead; i++) {
        buffer[offset + i] = {{{ makeGetValue('dataBuffer', 'i', 'i8')}}}
```
The `if` isn't needed, I think? The fix is the condition in the loop, With the new condition, it does the check in the `if` anyhow.